cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -285,95 +331,69 @@ func performEtcdStaticPodUpgrade(client clientset.Interface, waiter apiclient.Wa
 	if strings.Compare(desiredEtcdVersion.String(), currentEtcdVersion.String()) == 0 {
 		return false, nil
 	}
+	return true, nil
+}
 
-	beforeEtcdPodHash, err := waiter.WaitForStaticPodSingleHash(cfg.NodeRegistration.Name, constants.Etcd)
+// rolledBackUpgradeComponentFail is used to rolled back oldEtcd when newEtcd upgrade component failed
+func rolledBackUpgradeComponentFail(pathMgr StaticPodPathManager, cfg *kubeadmapi.InitConfiguration, oldEtcdClient etcdutil.ClusterInterrogator, backupEtcdDir string) error {
```

Comment:
This func probably needs to have `etcd` somewhere in its name. After all, this can be accessed from anywhere in the upgrade package and its name implies generic component recovery.